perception/traffic_light_map_based_detector/src/node.cpp

```diff
@@ -171,100 +231,136 @@ void MapBasedDetector::cameraInfoCallback(
    * Get the ROI from the lanelet and the intrinsic matrix of camera to determine where it appears
    * in image.
    */
+  /**
+   * set all offset to zero when calculating the expect roi
+   */
+  Config expect_roi_cfg = config_;
+  expect_roi_cfg.max_vibration_depth = 0;
+  expect_roi_cfg.max_vibration_height = 0;
+  expect_roi_cfg.max_vibration_width = 0;
+  expect_roi_cfg.max_vibration_yaw = 0;
+  expect_roi_cfg.max_vibration_pitch = 0;
   for (const auto & traffic_light : visible_traffic_lights) {
-    autoware_auto_perception_msgs::msg::TrafficLightRoi tl_roi;
+    autoware_auto_perception_msgs::msg::TrafficLightRoi rough_roi, expect_roi;
+    if (!getTrafficLightRoi(
+          tf_map2camera, pinhole_camera_model, traffic_light, expect_roi_cfg, expect_roi)) {
+      continue;
+    }
     if (!getTrafficLightRoi(
-          camera_pose_stamped.pose, pinhole_camera_model, traffic_light, config_, tl_roi)) {
+          tf_map2camera_vec, pinhole_camera_model, traffic_light, config_, rough_roi)) {
       continue;
     }
-    output_msg.rois.push_back(tl_roi);
+    output_msg.rois.push_back(rough_roi);
+    expect_roi_msg.rois.push_back(expect_roi);
   }
+
   roi_pub_->publish(output_msg);
-  publishVisibleTrafficLights(camera_pose_stamped, visible_traffic_lights, viz_pub_);
+  expect_roi_pub_->publish(expect_roi_msg);
+  cam_info_pub_->publish(*input_msg);
```

Comment:
It's necessary for the traffic_light_multi_camera_fusion package.
For the fusion, there are multiple topics needed to be subscribed for every camera, and it's trivial to configure every topic for every camera. So instead of setting the needed topics, it's easier to set only the camera namespaces. Then the topics can be adopted from the namespaces, as introduced [Here](https://github.com/Mingyu1991/autoware.universe/blob/feat/traffic_light_develop/perception/traffic_light_multi_camera_fusion/README.md?plain=1#L12). However, the original camera info topic is not in the namespace, so it has to be re-published.

Reply:
OK, I see.
For that reason, I think it's better to use relay tool (like [this](https://github.com/tier4/autoware_launch.xx1/blob/tier4/main/autoware_launch/launch/autoware.launch.xml#L124-L128)) outside of this node.

Reply:
fixed in commit 300b1dfd6d31f24058a6ac0349f828bf624ff66c